metadoc-js/src/main/scala/metadoc/MetadocApp.scala

```diff
@@ -135,6 +162,7 @@ object MetadocApp {
 
     for (editor <- editorService.open(input)) {
       updateTitle(state)
+      registerWorkspaceFiles(editor, workspace)
```

Comment:
I suggest this is only called once per editor, for example in `main` right after

```scala
val editorService = new MetadocEditorService(index)
```

This would also avoid having to pass it around everywhere.